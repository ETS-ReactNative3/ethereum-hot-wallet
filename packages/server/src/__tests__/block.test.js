// @flow
import micro from 'micro';
import listen from 'test-listen';
import request from 'request-promise';
import index from '../index';
import MOCK_BLOCK from '../__mock-data__/block';
import { type Handler, type Req, type Res } from '../utils/type.flow';

jest.mock('../utils/web3-helper', () => ({
  extend: () => {},
  middleware: () => (handler: Handler) => (
    req: Req,
    res: Res,
    ...restArgs: any
  ) => {
    req.context = {
      ...req.context,
      web3: {
        extend: () => {},
        eth: {
          getTransaction: () => Promise.resolve({}),
          getBlock: () => Promise.resolve(MOCK_BLOCK),
        },
        admin: {
          nodeInfo: () => Promise.resolve({}),
        },
      },
    };
    return handler(req, res, ...restArgs);
  },
}));

it('should return correct json data', async () => {
  const service = micro(index);
  let error;
  try {
    const url = await listen(service);
    const response = await request({
      uri: `${url}/block/13`,
    });
    expect(response).toMatchSnapshot();
  } catch (err) {
    error = err;
  }

  expect(error).toBeUndefined();
  service.close();
});
