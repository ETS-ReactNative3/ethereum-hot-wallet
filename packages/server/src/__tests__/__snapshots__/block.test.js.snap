// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should return correct json data 1`] = `"{\\"difficulty\\":\\"7724697433196\\",\\"gasLimit\\":3141592,\\"gasUsed\\":42000,\\"hash\\":\\"0xfb8580b170296d5a2c9e7897e68b240bb66d4e60c285bd7621a43dcb92d3e09c\\",\\"miner\\":\\"0x957523c697993cB24FCf75EfBce4f228325e7383\\",\\"parentHash\\":\\"0x342d528eb9c8a5b1fe7a10756573d8f218933930868295fcd93e03107f4edebf\\",\\"totalDifficulty\\":\\"3588404905402599053\\",\\"transactions\\":[\\"0x4b6776c238213a86fa01852f6faeec4f38a4ca5248b488166abc1efef407da06\\",\\"0xd6a80a88b9564aa7580b90ca2e5fec8da9d113f54370f86f72fcf0e40f05d131\\"]}"`;
